core/src/main/java/org/infinispan/topology/ClusterTopologyManagerImpl.java

```diff
@@ -502,7 +502,7 @@ private void recoverClusterStatus(int newViewId, final List<Address> clusterMemb
       CountDownLatch latch = new CountDownLatch(responsesByCache.size());
       LimitedExecutor cs = new LimitedExecutor("Merge-" + newViewId, stateTransferExecutor, maxThreads);
       for (final Map.Entry<String, Map<Address, CacheStatusResponse>> e : responsesByCache.entrySet()) {
-         CacheJoinInfo joinInfo = e.getValue().values().stream().findAny().get().getCacheJoinInfo();
+         CacheJoinInfo joinInfo = e.getValue().values().iterator().next().getCacheJoinInfo();
```

Comment:
I wish ordered collections had a first() method that encapsulates iterator().next()

Reply:
I don't think it's that bad, `first()` would probably return an `Optional` like the one in `Stream` so you'd need an extra method call anyway. The only advantage would be replacing the `NoSuchElementException` with our own, but then that would be more verbose, and we assume at least one response anyway.